src/mscorlib/src/System/Span.cs

```diff
@@ -625,5 +636,266 @@ internal static unsafe void CopyTo<T>(ref T destination, ref T source, int eleme
+                    i += 4;
+                }
+
+                nuint end = byteLength - 16;
+                byteLength -= i; // lower 4 bits of byteLength represent how many bytes are left *after* the unrolled loop
+
+                // We know due to the above switch-case that this loop will always run 1 iteration; max
+                // bytes we clear before checking is 23 (7 to align the pointers, 16 for 1 iteration) so
+                // the switch handles lengths 0-22.
+                Debug.Assert(end >= 7 && i <= end);
+
+                // This is separated out into a different variable, so the i + 16 addition can be
+                // performed at the start of the pipeline and the loop condition does not have
+                // a dependency on the writes.
+                nuint counter;
+
+                do
+                {
+                    counter = i + 16;
+
+                    // This loop looks very costly since there appear to be a bunch of temporary values
+                    // being created with the adds, but the jit (for x86 anyways) will convert each of
+                    // these to use memory addressing operands.
+
+                    // So the only cost is a bit of code size, which is made up for by the fact that
+                    // we save on writes to bytePointer.
+
+                    *(long*)(bytePointer + i) = (long)0;
+                    *(long*)(bytePointer + i + 8) = (long)0;
+
+                    i = counter;
+
+                    // See notes above for why this wasn't used instead
+                    // i += 16;
+                }
+                while (counter <= end);
+
+                if ((byteLength & 8) != 0)
+                {
+                    *(long*)(bytePointer + i) = (long)0;
+                    i += 8;
+                }
+                if ((byteLength & 4) != 0)
+                {
+                    *(int*)(bytePointer + i) = (int)0;
+                    i += 4;
+                }
+                if ((byteLength & 2) != 0)
+                {
+                    *(short*)(bytePointer + i) = (short)0;
+                    i += 2;
+                }
+                if ((byteLength & 1) != 0)
+                {
+                    *(bytePointer + i) = (byte)0;
+                    // We're not using i after this, so not needed
+                    // i += 1;
+                }
+
+                return;
+
+                PInvoke:
+                RuntimeImports.RhZeroMemory(ref b, byteLength);
+            }
+        }
+
+        internal static unsafe void ClearWithReferences(ref IntPtr ip, nuint pointerSizeLength)
+        {
+            if (pointerSizeLength == 0)
+                return;
+
+            // TODO: Perhaps do switch casing to improve small size perf
+
+            nuint i = 0;
+            nuint n = 0;
+            while ((n = i + 8) <= (pointerSizeLength))
+            {
+                Unsafe.Add<IntPtr>(ref ip, (int)i + 0) = default(IntPtr);
```

Comment:
Yes, it would work, but it will require IntPtr casts. Since you need to add it to CoreLib anyway you can as well add versions that just take nuint so that you do not need the casts.